test/suite/template/csTemplate.test.ts

```diff
@@ -0,0 +1,96 @@
+import * as assert from 'assert';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import * as vscode from 'vscode';
+
+import CsTemplate from '../../../src/template/csTemplate';
+// import { ExtensionError } from '../../../src/util';
+
+const templates_path = path.resolve(__dirname, '../../../../templates/');
+const fixture_path= path.resolve(__dirname, '../../suite/');
+const fixtureCsTemplate = 
+{
+    filename: 'cstemplate.csproj',
+    csproj: `
+    <Project Sdk="Microsoft.NET.Sdk">
+        <PropertyGroup>
+            <TargetFramework>net6.0</TargetFramework>
+        </PropertyGroup>
+    </Project>
+    `, 
+};
+const expectedEol = (eolSettings: string|undefined): string => {
+    switch (eolSettings) {
+        case '\n':
+            return '\n';
+        case '\r\n':
+            return '\r\n';
+        default:
+        case 'auto':
+            return os.EOL;
+    }
+};
+
+suite('CsTemplate', () => {
+    test('create cs template when wrong template path, throws exception', () => {
+        const csTemplate = new CsTemplate('test', 'createClass',);
+
+        return csTemplate.create('notExsistingTemplate', 'testfilePath', 'test')
+            .catch((err) => {
+                assert.strictEqual(err.message, 'Could not read template file from \'notExsistingTemplate/test.tmpl\'');
+            });
+    });
+
+    // FIXME: this test should fail with a different error
```

Comment:
@KreativJos This commented test hides an hidden bug. This `csTemplate.create(templates_path, '', '')` won't raise any kind of error, which seems to be wrong.

Reply:
Would be overkill refactoring the create method from here to something like TemplateFactory class?
The method almost the same in every *Template class except in CshtmlTemplate. A method inside the factory could iterate over the template.getExtensions() method, create the necessary files and the *Template classes would just store the raw template and for example a `generate()` method would return the final string that the factory class would save to disk.
This change would make the testing more easier (less setup, faster tests wihtout any IO).